### PR TITLE
Create yaml version of daily JS build

### PR DESCRIPTION
**Pull request by BruceHaley**
New build name: Botbuilder-JS-daily-yaml.

### CODE REVIEW

**Review by tsuwandy — APPROVED** [review]

File: build/yaml/js-build-steps.yml [hunk 1]
```diff
@@ -79,10 +84,6 @@ steps:
     verbose: false
     customCommand: 'run build'
 
-- task: azure-cosmosdb.emulator-public-preview.run-cosmosdbemulatorcontainer.CosmosDbEmulator@2
-  displayName: 'Run Azure Cosmos DB Emulator container'
-  enabled: false
-
```
Comment (stevengum) [inline build/yaml/js-build-steps.yml:85]: Was this ever run?

File: build/yaml/botbuilder-js-daily.yml [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+  displayName: 'Resolve package version variables'
+
+- template: js-build-steps.yml
+
+- task: Npm@1
+  displayName: 'npm run eslint'
+  inputs:
+    command: custom
+    verbose: false
+    customCommand: 'run eslint'
```
Comment (stevengum) [inline build/yaml/botbuilder-js-daily.yml:40]: I don't believe the pipeline should run `eslint` at this time, it seems a little "too little, too late". If we're cutting a daily we should have run eslint already in another pipeline.

File: build/yaml/botbuilder-js-daily.yml [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+    publishEndpoint: 'MyGet botbuilder JS Scratch Feed'
+
+- task: Npm@1
+  displayName: 'npm pack botbuilder'
+  inputs:
+    command: custom
+    workingDir: libraries/botbuilder
+    verbose: false
+    customCommand: pack
+    publishEndpoint: 'MyGet botbuilder JS Scratch Feed'
```
Comment (stevengum) [inline build/yaml/botbuilder-js-daily.yml:67]: Can you sort these `packs` to match the order in `libraries/`?

**Review by stevengum — CHANGES_REQUESTED** [review]

File: build/yaml/botbuilder-js-daily.yml [hunk 4]
```diff
@@ -0,0 +1,266 @@
+#
+# This runs under BotBuilder-JS-daily-yaml.
+#
+
+# "name" here defines the build number format. Build number is accessed via $(Build.BuildNumber)
+name: $(Build.BuildId)
+
+pool:
+  vmImage: 'macOS-10.15'
```
[the remark this answers is not in the document]
Comment (BruceHaley) [reply build/yaml/botbuilder-js-daily.yml:9]: Changing all pipelines to use ubuntu-18.04 from macOS-10.15.

File: build/yaml/botbuilder-js-daily.yml [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+variables:
+  NodeVersion: 10.x
```
[the remark this answers is not in the document]
Comment (BruceHaley) [reply build/yaml/botbuilder-js-daily.yml:12]: Changing it to 12.x

File: build/yaml/botbuilder-js-daily.yml (hunk 2 above)
Comment (BruceHaley) [reply build/yaml/botbuilder-js-daily.yml:40]: Moving eslint to js-build-steps.yml.

File: build/yaml/botbuilder-js-daily.yml (hunk 3 above)
Comment (BruceHaley) [reply build/yaml/botbuilder-js-daily.yml:67]: Good idea. Put in Windows folders order.

File: build/yaml/js-build-steps.yml (hunk 1 above)
Comment (BruceHaley) [reply build/yaml/js-build-steps.yml:85]: I don't think so.

File: build/yaml/botbuilder-js-daily.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (stevengum) [reply build/yaml/botbuilder-js-daily.yml:9]: Thanks for filing the issue, I've asked the other repo leads to chime in on it as well and provided some context.

File: build/yaml/botbuilder-js-daily.yml (hunk 2 above)
Comment (stevengum) [reply build/yaml/botbuilder-js-daily.yml]: Thanks for moving it here @BruceHaley, but let's just remove it from `js-build-steps.yml`.

I have eslint running in another workflow and the build ci isn't meant to fail status checks for unlinted files.

Comment (BruceHaley) [reply build/yaml/botbuilder-js-daily.yml]: Done.

File: build/yaml/botbuilder-js-daily.yml [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+   "Raw PreviewPackageVersion = $(PreviewPackageVersion)";
+   $ppv = "$(PreviewPackageVersion)".Replace("{DateStamp}",$dateStamp).Replace("{CommitHash}",$commitHash);
+   Write-Host "##vso[task.setvariable variable=PreviewPackageVersion;]$ppv";
+   "Resolved PreviewPackageVersion = $ppv";
+  displayName: 'Resolve package version variables'
+
+- template: js-build-steps.yml
+
+- task: Npm@1
+  displayName: 'npm pack adaptive-expressions'
```
Comment (joshgummersall) [inline build/yaml/botbuilder-js-daily.yml:36]: Can we use `lerna exec npm pack` to automatically pack everything referenced in the `lerna.json` file? Single source of truth would be nice 👍

Comment (BruceHaley) [reply build/yaml/botbuilder-js-daily.yml:36]: Good idea. That is outside the scope of this PR, so I have created Issue #2908 for that and assigned it to myself. https://github.com/microsoft/botbuilder-js/issues/2908

**Review by joshgummersall — APPROVED** [review]
Looks good, thanks!

**Pull request merged**